File: day01/day01.c
#include "day01.h"

int solve_part_1(char *input, size_t length) {
    int floor = 0;

    for (size_t i = 0; i < length; i++) {
        switch (input[i]) {
        case '(':
            floor++;
            break;
        case ')':
            floor--;
            break;
        }
    }

    return floor;
}

int solve_part_2(char *input, size_t length) {
    int floor = 0;

    for (size_t i = 0; i < length; i++) {
        switch (input[i]) {
        case '(':
            floor++;
            break;
        case ')':
            floor--;
            break;
        }

        if (floor < 0) {
            return i + 1;
        }
    }

    return 0;
}

File: src/day01.py
import sys


def solve_part_1(line: str) -> int:
    floor = 0
    for char in line:
        if char == "(":
            floor += 1
        if char == ")":
            floor -= 1
    return floor


def solve_part_2(line: str) -> int:
    floor = 0
    for idx, char in enumerate(line):
        if char == "(":
            floor += 1
        if char == ")":
            floor -= 1
        if floor < 0:
            return idx + 1
    return -1


if __name__ == "__main__":
    with open(sys.argv[1], "r") as f:
        line = f.readline()
        print(solve_part_1(line))
        print(solve_part_2(line))
